DynamiteClothing.py

```diff
@@ -0,0 +1,68 @@
+import scrapy
+from scrapy.spiders import CrawlSpider, Rule
+from scrapy.linkextractors import LinkExtractor
+
+from dynamiteclothing.items import Product
+
+
+class DynamiteclothingSpider(CrawlSpider):
+    name = 'DynamiteClothing'
+    allowed_domains = ['www.dynamiteclothing.com']
+
+    rules = (
+        Rule(LinkExtractor(restrict_css=[
+            '.subcategory.subCatLink', '#catPageNext'])),
+        Rule(LinkExtractor(
+            restrict_css='[title="View Product Details"]'),
+            callback='parse_product')
+    )
+
+    def start_requests(self):
+        yield scrapy.Request(
+            'https://www.dynamiteclothing.com/us/profile/login.jsp?h=true',
+            callback=self.generate_login_request)
+
+    def generate_login_request(self, response):
+        return scrapy.FormRequest.from_response(
+            response,
+            formid='loginForm',
+            formdata={
+                'loginEmail': 'zain@gmail.com',
+                '/atg/userprofiling/ProfileFormHandler.value.password': 'ABc123'},
+            callback=self.after_login
+        )
+
+    def after_login(self, response):
```

Comment:
function name should be a verb and as per scrapy recommendation `parse` should be a prefix for callback function.